oneview_redfish_toolkit/api/network_port.py

```diff
@@ -42,30 +40,13 @@ def __init__(self, device_id, port_id, server_hardware):
                 server_hardware: Oneview's server hardware dict
         """
         super().__init__(self.SCHEMA_NAME)
-        index = int(device_id) - 1
-        # port_id validation
-        try:
-            port_index = -1
-            port_count = -1
-            for port in server_hardware["portMap"]["deviceSlots"][index][
-                "physicalPorts"]:
-                port_count += 1
-                if port["portNumber"] == int(port_id):
-                    if port["type"] not in [
-                        "Ethernet", "FibreChannel", "Infiniband"]:
-                        raise OneViewRedfishError(
-                            "Port id refers to invalid port type")
-                    port_index = port_count
-                    break
-            if port_index == -1:
-                raise OneViewRedfishResourceNotFoundError(
-                    port_id, "NetworkPort")
-        except Exception:
-            raise OneViewRedfishResourceNotFoundError(
-                port_id, "NetworkPort")
 
-        port = server_hardware["portMap"]["deviceSlots"][index][
-            "physicalPorts"][port_index]
+        physical_ports = self.get_resource_by_id(
```

Comment:
you killed the tests made at lines 54-59. you need to validate port type.

Reply:
Fixed.